### PR TITLE
Update iceberg to document truncation and view creation

### PR DESCRIPTION
**Pull request by esemeniuc**

### CODE REVIEW

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -196,3 +200,8 @@ and a file system location of ``/var/my_tables/test_table``::
         format = 'PARQUET',
         partitioning = ARRAY['c1', 'c2'],
         location = '/var/my_tables/test_table')
+
+Iceberg limitations
+-------------------
+
+Iceberg cannot be used to create views. Details `here <https://github.com/trinodb/trino/blob/e4cae791c0b56509af281ceb7b91bc7c62dce6e5/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L1035>`_.
```
Comment (raunaqmorarka) [inline docs/src/main/sphinx/connector/iceberg.rst:207]: We generally don't link to the code in the documentation
It is sufficient to say that `Iceberg does not support creation of VIEWs`

Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst:207]: Agreed. Also dont use "here" for links. Along these lines we will work towards a doc PR to add a "SQL support" section in the connector docs that detail what actually is supported.

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -196,3 +200,8 @@ and a file system location of ``/var/my_tables/test_table``::
         format = 'PARQUET',
         partitioning = ARRAY['c1', 'c2'],
         location = '/var/my_tables/test_table')
+
+Iceberg limitations
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:204]: Also remove Iceberg from this title .. just 

Limitations

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 1 above)
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.rst:207]: Should this be sth like

```
Limitations
-----------

The following SQL statements are not yet supported:

* :doc:`/sql/create-view`
```

?

Comment (esemeniuc) [reply docs/src/main/sphinx/connector/iceberg.rst]: Updated the text

Comment (esemeniuc) [reply docs/src/main/sphinx/connector/iceberg.rst]: Changed now

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -196,3 +199,12 @@ and a file system location of ``/var/my_tables/test_table``::
...
         location = '/var/my_tables/test_table')
+
+Limitations
+-----------
+
+The following SQL statements are not yet supported:
+
+* :doc:`/sql/create-view`
+  
+  - `Github Issue <https://github.com/trinodb/trino/issues/8216>`_
```
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.rst:210]: remove please

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 4]
```diff
@@ -96,7 +96,10 @@ Transform                             Description
                                       partition value is an integer hash of ``x``, with a value between
                                       0 and ``nbuckets - 1`` inclusive.
 
-``truncate(s, nchars)``               The partition value is the first ``nchars`` characters of ``s``.
+``truncate(s, w)``                    If ``s`` is a string, the partition value is the first ``w`` 
+                                      characters of ``s`` for strings. If ``s`` is an integer, the 
+                                      partition value is rounded to the largest multiple of ``w`` less <=
+                                      to ``s``. `More details <https://iceberg.apache.org/spec/#truncate-transform-details>`_
```
Comment (findepi) [inline docs/src/main/sphinx/connector/iceberg.rst:102]: replace `<=` with "less than or equal"

does `truncate(-197, 10)` return `-200` or `-190`?

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 5]
```diff
@@ -196,3 +199,12 @@ and a file system location of ``/var/my_tables/test_table``::
         format = 'PARQUET',
         partitioning = ARRAY['c1', 'c2'],
         location = '/var/my_tables/test_table')
+
+Limitations
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:203]: Remove this whole Limitations sections .. I will submit a PR shortly that adds a `SQL support` section and it we will link to all supported statements and add that no other statements are supported. I will work with this next week with @Jessie212

Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst:203]: This is now implemented and the limitation no longer applies.

**Comment by raunaqmorarka** [comment]
@esemeniuc @mosabua @hashhar should this be closed now or does this still add useful documentation ?

**Comment by hashhar** [comment]
IMO we should directly link to Iceberg docs for explaining the bucket functions instead of keeping a static doc on our side (since the Iceberg spec can evolve over time).

I think this PR needs to be replaced with a new one. cc: @mosabua

**Comment by esemeniuc** [comment]
Please feel free to fork/drop this, currently not working with Iceberg

**Pull request closed without merging**